subsys/shell/shell.c

```diff
@@ -1280,22 +1280,14 @@ static void shell_signal_handle(const struct shell *sh,
 				enum shell_signal sig_idx,
 				shell_signal_handler_t handler)
 {
-<<<<<<< HEAD
 	struct k_poll_signal *sig = &sh->ctx->signals[sig_idx];
-=======
-	struct k_poll_signal *signal = &sh->ctx->signals[sig_idx];
->>>>>>> shell: fix MISRA 5.7 violations on `struct shell`
 	int set;
 	int res;
 
 	k_poll_signal_check(sig, &set, &res);
 
 	if (set) {
-<<<<<<< HEAD
 		k_poll_signal_reset(sig);
-=======
-		k_poll_signal_reset(signal);
->>>>>>> shell: fix MISRA 5.7 violations on `struct shell`
```

Comment:
Please drop last two commits.

Reply:
I apologize for my lack of understanding about git. I think my last two commits were in response to the need to rebase my branch onto the main branch. What would be the best way to drop them, clean up the git history, and still have it able to merge into main?

Reply:
IMO the best way for you is to squash into one commit, e.g. use `git rebase -i HEAD~4` and edit the list,

```
pick b2c1dfd9f39 Shell: Add get return code command.
pick 75a7a1d6e4c Shell: add retrieve return value via shell API
pick d5a51e0b3b2 shell: fix MISRA 5.7 violations on `struct shell`
pick 6e710d90027 Shell: add retrieve return value via shell API

# Rebase von 546ea7f977f..6e710d90027 auf 546ea7f977f (4 Kommandos)
```

to be

```
pick b2c1dfd9f39 Shell: Add get return code command.
s 75a7a1d6e4c Shell: add retrieve return value via shell API
s d5a51e0b3b2 shell: fix MISRA 5.7 violations on `struct shell`
s 6e710d90027 Shell: add retrieve return value via shell API

# Rebase von 546ea7f977f..6e710d90027 auf 546ea7f977f (4 Kommandos)
```

fix the commit message, git log:
```

commit 8f85799bc8e74bf16b4dd986707b6edf3598774c (HEAD -> 56130)
Author: Hunter Searle <hsearle@xes-inc.com>
Date:   Wed Mar 22 16:20:46 2023 -0500

    Shell: Add get return code command.
    
    This new commmand, and corresponding update to command
    execution, enables the user to retrieve the return code
    of the most recently run shell command. This is useful both
    for users wanting specific information beyond the print
    statements in a command, but especially for automated users
    that prefer a simple return code to parsing human readable text.
    This was tested using all default shell commands, as well as
    eeprom, flash, and i2c, where specific errors could be forced.
    In all cases, the correct return value was returned by the new
    retval command.
    
    Signed-off-by: Hunter Searle <hsearle@xes-inc.com>

commit 546ea7f977f278b2eb212013062bc040c6dc0faa
....

```

optional check the diff to you remove branch (git diff refs/pull/upstream/56130), and force push.

https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt---interactive

Reply:
Thanks for the guidance.

Reply:
Okay, in any case, there is a backup for you https://github.com/jfischer-no/zephyr/commit/8f85799bc8e74bf16b4dd986707b6edf3598774c.patch